monai/transforms/spatial/functional.py

```diff
@@ -295,10 +295,12 @@ def resize(img, out_size, mode, align_corners, dtype, input_ndim, anti_aliasing,
         lazy_evaluation=transform_info.get(TraceKeys.LAZY_EVALUATION, False),
     )
     out = convert_to_tensor(img.as_tensor() if isinstance(img, MetaTensor) else img, track_meta=get_track_meta())
-    if transform_info.get(TraceKeys.LAZY_EVALUATION, False) or tuple(convert_to_numpy(orig_size)) == out_size:
+    if transform_info.get(TraceKeys.LAZY_EVALUATION, False):
         if anti_aliasing and transform_info.get(TraceKeys.LAZY_EVALUATION, False):
             warnings.warn("anti-aliasing is not compatible with lazy evaluation.")
         return out.copy_meta_from(meta_info) if isinstance(out, MetaTensor) else meta_info
+    if tuple(convert_to_numpy(orig_size)) == out_size:
+        return out.copy_meta_from(meta_info) if isinstance(out, MetaTensor) else out
```

Comment:
to be consistent with line 321, this should always return float32...

also there's the same issue here https://github.com/Project-MONAI/MONAI/blob/9fd6d4ced3efb344cf3b24c51d899d8fafc882d5/monai/transforms/spatial/functional.py#L135-L138

I'll update